### PR TITLE
guidelines: add repository

### PR DESCRIPTION
**Pull request by 0xc0170**
Each repo should contain license & contributing information

I added it  to the guidelines (new section - new repository). This should be valid for any repository we release. 3 files that are important for contribution.

@ARMmbed/mbed-os-maintainers Please review

### CODE REVIEW

**Comment by 0xc0170** [comment]
Note, Mbed OS repository follows this.  Not all examples but at least some new ones do. 
This would be really nice to have in the documentation. Any new repository should conform (the old ones will be done via script later, in 5.12 most likely)

**Comment by 0xc0170** [comment]
I fixed the links

File: docs/reference/contributing/guidelines/guidelines.md [hunk 1]
```diff
@@ -19,3 +19,11 @@ If so, you can request to become an organization member, but you must first ensu
 - All users must have 2 Factor Authentication enabled.
 - Arm staff must have their Name, Company (Arm), Location and Arm email address publicly visible.
 - All others should have their Name and Company visible. Entering your Location will help us interpret response times according to time zones.
+
+### New repository
+
+Each repository must contain:
+
+- `CONTRIBUTING.md` file (similar to [Mbed OS contributing](https://github.com/ARMmbed/mbed-os/blob/master/CONTRIBUTING.md)).
+- `LICENSE` file - the full license text or overview of every license in the repo with links.
```
Comment (cmonr) [inline docs/reference/contributing/guidelines/guidelines.md]: Might not be a question for this PR, but do we need to specify which kind of licenses should be used?

Comment (0xc0170) [reply docs/reference/contributing/guidelines/guidelines.md]: Shall we add a link here to license page ?

Comment (cmonr) [reply docs/reference/contributing/guidelines/guidelines.md]: Oh, yes!

Comment (0xc0170) [reply docs/reference/contributing/guidelines/guidelines.md]: `* Licenses: Licenses should comply with the [licenses described here](../contributing/license.html).` is above in the first section, should be sufficient for this page

**Comment by 0xc0170** [comment]
One thing I noticed - our repos ar emissing templates for issues/pull requests. As a guidance, we might want to document these two. See an example here : https://github.com/ARMmbed/mbed-os-example-mbed-crypto/pull/5

I can send separate PR based on this one to extend what repository should follow. These templates should be a must / should contain. So our scripts work (cloning issues), etc

**Pull request merged**